### PR TITLE
Massive Cleanup of YourHome Markup

### PR DESCRIPTION
**Pull request by duke7553**
With this PR, we do the following:
- Remove unused styles and templates from the markup
- Tweak the design by removing unnecessary/subpar implementations of depth and color
- Introduce the concept of a dialog box helper and make every warning message use it for consistency

image

### CODE REVIEW

File: Files/Interacts/Interaction.cs [hunk 1]
```diff
@@ -435,8 +433,7 @@ private async void OpenSelectedItems(bool displayApplicationPicker)
             }
             catch (FileNotFoundException)
             {
-                MessageDialog dialog = new MessageDialog("The file you are attempting to access may have been moved or deleted.", "File Not Found");
-                await dialog.ShowAsync();
+                await DialogDisplayHelper.ShowDialog("File Not Found", "The file you are attempting to access may have been moved or deleted.");
```
Comment (yaichenbaum) [inline Files/Interacts/Interaction.cs:390]: We should use localized strings here.

File: Files/Interacts/Interaction.cs [hunk 2]
```diff
@@ -676,12 +673,11 @@ public async void DeleteItem_Click(object sender, RoutedEventArgs e)
...
             catch (UnauthorizedAccessException)
             {
-                MessageDialog AccessDeniedDialog = new MessageDialog("Access Denied", "Unable to delete this item");
-                await AccessDeniedDialog.ShowAsync();
+                await DialogDisplayHelper.ShowDialog("Access Denied", "We weren't able to delete this item");
             }
             catch (FileNotFoundException)
             {
-                Debug.WriteLine("Attention: Tried to delete an item that could be found");
+                await DialogDisplayHelper.ShowDialog("Item Not Found", "We weren't able to delete this item because it cannot be found.");
```
Comment (yaichenbaum) [inline Files/Interacts/Interaction.cs:561]: We should use localized strings here.

File: Files/Interacts/Interaction.cs [hunk 3]
```diff
@@ -676,12 +673,11 @@ public async void DeleteItem_Click(object sender, RoutedEventArgs e)
             }
             catch (UnauthorizedAccessException)
             {
-                MessageDialog AccessDeniedDialog = new MessageDialog("Access Denied", "Unable to delete this item");
-                await AccessDeniedDialog.ShowAsync();
+                await DialogDisplayHelper.ShowDialog("Access Denied", "We weren't able to delete this item");
```
Comment (yaichenbaum) [inline Files/Interacts/Interaction.cs:557]: We should use localized strings here.

File: Files/Interacts/Interaction.cs [hunk 4]
```diff
@@ -1164,9 +1160,7 @@ public async void ToggleQuickLook()
             }
             catch (FileNotFoundException)
             {
-                MessageDialog dialog = new MessageDialog("The file you are attempting to preview may have been moved or deleted.", "File Not Found");
-                var task = dialog.ShowAsync();
-                task.AsTask().Wait();
+                await DialogDisplayHelper.ShowDialog("File Not Found", "The file you are attempting to preview may have been moved or deleted.");
```
Comment (yaichenbaum) [inline Files/Interacts/Interaction.cs:953]: We should use localized strings here.

File: Files/Helpers/DialogDisplayHelper.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        /// </param>
+        public static async Task<bool> ShowDialog(string title, string message, string primaryText = "OK", string secondaryText = null)
+        {
+            bool result = false;
+
+            try
+            {
+                if (Window.Current.Content is Frame rootFrame)
+                {
+                    var d = new ContentDialog
```
Comment (yaichenbaum) [inline Files/Helpers/DialogDisplayHelper.cs:37]: Can we use a more descriptive name instead of `d`?

File: Files/Helpers/DialogDisplayHelper.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+                        PrimaryButtonText = primaryText
+                    };
+
+                    if (!string.IsNullOrEmpty(secondaryText))
+                    {
+                        d.SecondaryButtonText = secondaryText;
+                    }
+                    var dr = await d.ShowAsync();
+
+                    result = (dr == ContentDialogResult.Primary);
```
Comment (yaichenbaum) [inline Files/Helpers/DialogDisplayHelper.cs:50]: Can we use a more descriptive name instead of `dr`?

File: Files/UserControls/YourHome.xaml.cs [hunk 7]
```diff
@@ -336,18 +234,46 @@ private void MenuFlyoutItem_Click(object sender, RoutedEventArgs e)
...
+            }
+
+            switch (App.AppSettings.LayoutMode)
+            {
+                case 0:
+                    App.CurrentInstance.ContentFrame.Navigate(typeof(GenericFileBrowser), NavigationPath); // List View
+                    break;
+
+                case 1:
+                    App.CurrentInstance.ContentFrame.Navigate(typeof(PhotoAlbum), NavigationPath); // Grid View
```
Comment (yaichenbaum) [inline Files/UserControls/YourHome.xaml.cs:272]: I think there will be a conflict here with the latest changes to layout modes.

File: Files/View Models/ItemViewModel.cs [hunk 8]
```diff
@@ -737,15 +735,13 @@ public async Task RapidAddItemsToCollectionAsync(string path)
             }
             catch (FileNotFoundException)
             {
-                MessageDialog folderGone = new MessageDialog("The folder you've navigated to was not found.", "Did you delete this folder?");
-                await folderGone.ShowAsync();
+                await DialogDisplayHelper.ShowDialog("Did you delete this folder?", "The folder you are attempting to access may have been moved or deleted.");
```
Comment (yaichenbaum) [inline Files/View Models/ItemViewModel.cs:738]: We should use localized strings here.

File: Files/View Models/ItemViewModel.cs [hunk 9]
```diff
@@ -737,15 +735,13 @@ public async Task RapidAddItemsToCollectionAsync(string path)
...
                 IsLoadingItems = false;
                 return;
             }
             catch (Exception e)
             {
-                MessageDialog driveGone = new MessageDialog(e.Message, "Did you unplug this drive?");
-                await driveGone.ShowAsync();
+                await DialogDisplayHelper.ShowDialog("Did you unplug this drive?", e.Message);
```
Comment (yaichenbaum) [inline Files/View Models/ItemViewModel.cs:744]: We should use localized strings here.

File: Files/UserControls/YourHome.xaml.cs (hunk 7 above)
Comment (duke7553) [reply Files/UserControls/YourHome.xaml.cs:272]: Good catch 😸🎖

**Pull request merged**